airflow/utils/dag_processing.py

```diff
@@ -63,204 +62,109 @@ class SimpleDag(BaseDag):
-        else:
-            self._pool = None
-        if hasattr(ti, 'priority_weight'):
-            self._priority_weight = ti.priority_weight
-        else:
-            self._priority_weight = None
-        self._queue = ti.queue
-        self._key = ti.key
-
-    @property
-    def dag_id(self):
-        return self._dag_id
-
-    @property
-    def task_id(self):
-        return self._task_id
-
-    @property
-    def execution_date(self):
-        return self._execution_date
-
-    @property
-    def start_date(self):
-        return self._start_date
-
-    @property
-    def end_date(self):
-        return self._end_date
-
-    @property
-    def try_number(self):
-        return self._try_number
-
-    @property
-    def state(self):
-        return self._state
-
-    @property
-    def pool(self):
-        return self._pool
-
-    @property
-    def priority_weight(self):
-        return self._priority_weight
-
-    @property
-    def queue(self):
-        return self._queue
-
-    @property
-    def key(self):
-        return self._key
-
-    @property
-    def executor_config(self):
-        return self._executor_config
-
-    @provide_session
-    def construct_task_instance(self, session=None, lock_for_update=False):
-        """
-        Construct a TaskInstance from the database based on the primary key
-
-        :param session: DB session.
-        :param lock_for_update: if True, indicates that the database should
-            lock the TaskInstance (issuing a FOR UPDATE clause) until the
-            session is committed.
-        """
-        TI = airflow.models.TaskInstance
-
-        qry = session.query(TI).filter(
-            TI.dag_id == self._dag_id,
-            TI.task_id == self._task_id,
-            TI.execution_date == self._execution_date)
-
-        if lock_for_update:
-            ti = qry.with_for_update().first()
-        else:
-            ti = qry.first()
-        return ti
-
-
 class SimpleDagBag(BaseDagBag):
     """
     A collection of SimpleDag objects with some convenience methods.
     """
 
-    def __init__(self, simple_dags):
+    def __init__(self, simple_dags: List[SimpleDag]):
         """
         Constructor.
 
         :param simple_dags: SimpleDag objects that should be in this
-        :type list(airflow.utils.dag_processing.SimpleDagBag)
+        :type List(airflow.utils.dag_processing.SimpleDag)
         """
         self.simple_dags = simple_dags
-        self.dag_id_to_simple_dag = {}
+        self.dag_id_to_simple_dag: Dict[str, SimpleDag] = {}
 
         for simple_dag in simple_dags:
             self.dag_id_to_simple_dag[simple_dag.dag_id] = simple_dag
 
     @property
-    def dag_ids(self):
+    def dag_ids(self) -> KeysView[str]:
         """
         :return: IDs of all the DAGs in this
         :rtype: list[unicode]
         """
         return self.dag_id_to_simple_dag.keys()
 
-    def get_dag(self, dag_id):
+    def get_dag(self, dag_id: str) -> SimpleDag:
```

Comment:
Same here. `:rtype: airflow.utils.dag_processing.SimpleDag` can be removed, right?

Reply:
No not really - type hints and docs are sadly separate right now.

Reply:
> Nope. I added sphinx-autodoc-typehint that will add those automatically. I will review all the types in the affected files and see if there are any :type left.

https://github.com/apache/airflow/pull/6596#discussion_r351287732 I would prefer doing it like this.

Reply:
Yeah. I am testing if it *Really* works as advertised now.

Reply:
Unfortunately autodoc won't work with autoapi-type-hints extensions. I will leave both for now in the future we would like need to change api generation to autodoc :( :( very bad.